### PR TITLE
Fix mol2 with no bonds

### LINKED ISSUE #3057: MOL2 Reader and Parser don't accept files without bonds
## Expected behavior ##

Reading a MOL2 file that doesn't have any bonds (with only Na+ and Cl- atoms for instance) should be fine, as it is the case for PDB files.

## Actual behavior ##

It raises an Exception: `The mol2 block has no bonds`

The problem concerns both the MOL2Parser and MOL2Reader:
https://github.com/MDAnalysis/mdanalysis/blob/cf75dd6fe20e935c95fef57b0029382883ab0283/package/MDAnalysis/topology/MOL2Parser.py#L129-L131
https://github.com/MDAnalysis/mdanalysis/blob/cf75dd6fe20e935c95fef57b0029382883ab0283/package/MDAnalysis/coordinates/MOL2.py#L192-L194

I think those exceptions should both be removed.
Finally in the MOL2Parser, the Bonds attribute should only be added if bonds are present (just like in the PDBParser with #2832). Basically adding an `if bonds:` before this line:
https://github.com/MDAnalysis/mdanalysis/blob/cf75dd6fe20e935c95fef57b0029382883ab0283/package/MDAnalysis/topology/MOL2Parser.py#L183

And for the MOL2Writer, just writing an empty `@<TRIPOS>BOND` section if `obj.bonds` doesn't exist should do it?
https://github.com/MDAnalysis/mdanalysis/blob/cf75dd6fe20e935c95fef57b0029382883ab0283/package/MDAnalysis/coordinates/MOL2.py#L327-L338

I'll make a PR next week unless someone wants to give it a go before that

### PR DESCRIPTION
**Pull request by cbouy**
Fixes #3057 

Changes made in this Pull Request:
 - MOL2 files without bonds can now be read and written

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Review by richardjgowers — APPROVED** [review]
Lgtm thanks @cbouy

**Review by lilyminium — CHANGES_REQUESTED** [review]
Thank you @cbouy! Could you please just fix the pep8 notes <3

**Review by IAlibay — APPROVED** [review]
overall lgtm, couple of comments. I'll let @lilyminium approve & merge.

File: package/MDAnalysis/coordinates/MOL2.py [hunk 1]
```diff
@@ -322,20 +320,24 @@ def encode_block(self, obj):
...
-        bond_lines.append("\n")
-        bond_lines = "\n".join(bond_lines)
+        # only write bonds if the Bonds attribute exists (Issue #3057)
+        if hasattr(obj, "bonds"):
+            # Grab only bonds between atoms in the obj
+            # ie none that extend out of it
+            bondgroup = obj.bonds.atomgroup_intersection(obj, strict=True)
+            bonds = sorted((b[0], b[1], b.order) for b in bondgroup)
+            bond_lines = ["@<TRIPOS>BOND"]
+            bls = ["{0:>5} {1:>5} {2:>5} {3:>2}".format(bid,
```
[the remark this answers is not in the document]
Comment (cbouy) [reply package/MDAnalysis/coordinates/MOL2.py:330]: I thought about it but it's slightly too long to fit on one line

File: package/MDAnalysis/coordinates/MOL2.py [hunk 2]
```diff
@@ -322,20 +320,24 @@ def encode_block(self, obj):
...
+                                                        mapping[atom1],
+                                                        mapping[atom2],
+                                                        order)
+                   for bid, (atom1, atom2, order) in enumerate(bonds, start=1)]
+            bond_lines.extend(bls)
+            bond_lines.append("\n")
+            bond_lines = "\n".join(bond_lines)
+        else:
+            bondgroup = []
+            bond_lines = "@<TRIPOS>BOND\n"
```
[the remark this answers is not in the document]
Comment (cbouy) [reply package/MDAnalysis/coordinates/MOL2.py:340]: I just checked, it's a "free format" so not writing anything seems like the way to go. I'll change that

File: package/MDAnalysis/coordinates/MOL2.py [hunk 3]
```diff
@@ -127,6 +127,8 @@ class MOL2Reader(base.ReaderBase):
     .. versionchanged:: 0.20.0
        Allows for comments at top of file.
        Ignores status bit strings
+    .. versionchanged:: 2.0.0
```
[the remark this answers is not in the document]
Comment (cbouy) [reply package/MDAnalysis/coordinates/MOL2.py:130]: built fine on my end

[the remark this answers is not in the document]
Comment (IAlibay) [reply package/MDAnalysis/coordinates/MOL2.py:130]: cool, thanks for checking 👍

**Review by lilyminium — APPROVED** [review]

**Comment by lilyminium** [comment]
Thank you @cbouy!!

**Pull request merged**